option cc_generic_services = true;

message OpenRequest
{
    optional string db_name = 1;
}

message OpenResponse
{
}

message GetRequest
{
    optional string key = 1;
}

message GetResponse
{
    optional string key = 1;
    optional bytes value = 2;
}

message PutRequest
{
    optional string key = 1;
    optional bytes value = 2;
}

message PutResponse
{
}

message DeleteRequest
{
    optional string key = 1;
}

message DeleteResponse
{
}

message CloseRequest
{
}

message CloseResponse
{
}

service NetDBService
{
    rpc Open(OpenRequest) returns(OpenResponse);
    rpc Get(GetRequest) returns(GetResponse);
    rpc Put(PutRequest) returns(PutResponse);
    rpc Delete(DeleteRequest) returns(DeleteResponse);
    rpc Close(CloseRequest) returns(CloseResponse);
}
